src/plugins/data_source/server/types.ts

```diff
@@ -19,7 +19,7 @@ export interface LegacyClientCallAPIParams {
 }
 
 export interface DataSourceClientParams {
-  dataSourceId: string;
+  dataSourceId?: string;
```

Comment:
Just leaving a comment reminding us id is only optional for test client

Reply:
Added